contracts/HATKlerosV2Connector.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+
+/**
+ *  @authors: [@unknownunknown1]
+ *  @reviewers: []
+ *  @auditors: []
+ *  @bounties: []
+ *  @deployments: []
+ */
+
+pragma solidity 0.8.16;
+
+import "./HATArbitrator.sol"; // TODO: add interface
+import "./interfaces/IHATVault.sol";
+import { IArbitrable, IArbitrator } from "@kleros/erc-792/contracts/IArbitrator.sol";
+
+/**
+ *  @title HATKlerosV2Connector
+ *  @dev This contract acts a connector between HatsFinance and Kleros court V2. The contract doesn't support appeals and evidence
+ *  submisstion, since it'll be handled by the court.
+ */
+contract HATKlerosV2Connector is IArbitrable {
+
+    uint256 private constant RULING_OPTIONS = 2; // The amount of non 0 choices the arbitrator can give.
+  
+    enum Decision {
+        None, // Court wasn't able to make a decisive ruling. In this case the claim is executed. Both sides will get their appeal deposits back in this case.
+        ExecuteResolution, // Accept the claim as it is without changing.
+        DismissResolution // Dismiss the claim.
+    }
+
+    struct DisputeStruct {
+        bytes32 claimId; // Id of the claim in HATVault contract.
+        uint256 externalDisputeId; // Id of the dispute created in Kleros court.
+        Decision ruling; // Ruling given by the arbitrator.
+        bool resolved; // True if the dispute has been resolved.
+    }
```

Comment:
You could move this to a shared interface/ base contract with the V1 connector to avoid duplication